torch/_decomp/decompositions.py

```diff
@@ -580,6 +581,38 @@ def slice_backward(
     grad_input = grad_output.new_zeros(input_sizes)
     return torch.slice_scatter(grad_input, grad_output, dim, start, end, step)
 
+@register_decomposition(aten.slice.Tensor)
+def slice(
+# Tensor(a) self, int dim=0, SymInt? start=None, SymInt? end=None, SymInt step=1
+    self: Tensor,
+    dim: int = 0,
+    start: Optional[int] = None,
+    end: Optional[int] = None,
+    step:  Optional[int] = 1,
+):
+    new_size = list(self.size())
+    new_stride = list(self.stride())
+    new_offset = self.storage_offset()
+    dimsize = self.size()[dim]
+    if not start:
+        start = 0
+    if not end:
+        end = dimsize
+
+    canonical_start = max(min(start, dimsize), -dimsize) % dimsize
```

Comment:
so, what are the allowed set of operations to perform on a symint? and should i consider adding one if i wanted to or should i consider it already a principled set?

(currently:
```
    canonical_start = max(min(start, dimsize), -dimsize) % dimsize
TypeError: bad operand type for unary -: 'torch._C.SymIntNode'
```

Reply:
why don't you just do a bunch of conditionals, like how the C++ implementation used to do it? Those would have worked